### ISSUE TITLE
Fork CMS + Doctrine + \DateTime

### ISSUE THREAD

**Issue by jeroendesloovere** [forkcms/forkcms#1466]
### Example

I want to save $publishOn to "11/05/2016 20:00", this will be saved in the database as "2016-05-11 18:00:00" (Which is correct. Note the two hour difference, because of UTC+1 (Europe/Brussels) + summer time)

If you then want to show "H:i" in your Backend Edit.php action:
```php
date('H:i', $blogArticle->getPublishOn()->getTimestamp());
```
> This shows "18:00" in the website. Which is false!!! Because It should show "20:00"...!!!

### Actual problem

SpoonDate + SpoonForm

### Problems

**1) Doctrine always uses default timezone**
Doctrine always uses the default timezone that is in your setup ( "Europe/Brussels" in my case).

When saving
* A \DateTime, the timezone is NOT saved in the database.

So when loading:
* The \DateTime from the database, your default timezone ("Europe/Brussels") will be used. That is not what we want, it has to be "UTC".

**2) Fork CMS is actually not very good for international \DateTime**

Since we actually set ```date_default_timezone_set('Europe/Brussels');``` in the [`/src/Common/Core/Init.php`](https://github.com/forkcms/forkcms/blob/master/src/Common/Core/Init.php#L47).

### Solution recommended by Doctrine

> Always saving the timezone in your entity. [Read more](http://doctrine-orm.readthedocs.io/projects/doctrine-orm/en/latest/cookbook/working-with-datetime.html)

```php
<?php
namespace Shipping;

/**
 * @Entity
 */
class Event
{
    /** @Column(type="datetime") */
    private $created;

    /** @Column(type="string") */
    private $timezone;

    /**
     * @var bool
     */
    private $localized = false;

    public function __construct(\DateTime $createDate)
    {
        $this->localized = true;
        $this->created = $createDate;
        $this->timezone = $createDate->getTimeZone()->getName();
    }

    public function getCreated()
    {
        if (!$this->localized) {
            $this->created->setTimeZone(new \DateTimeZone($this->timezone));
        }
        return $this->created;
    }
}
```

### Another working solution

1) At [`/src/Common/Core/Init.php`](https://github.com/forkcms/forkcms/blob/master/src/Common/Core/Init.php#L47) setting `date_default_timezone_set('UTC')` without an if statement

2) Then you need to set the timezone in the backend Edit action:
```php
$this->frm->addTime(
    'publish_on_time',
    date(
        'H:i',
        $this->blogArticle->getPublishOn()->setTimeZone(
            new \DateTimeZone('Europe/Brussels')
        )->getTimestamp()
    )
);
```

### Other solutions?

* When defining Doctrine, maby we can set it to use timezone "UTC" instead of letting Doctrine get the default timezone.

**Comment by Thijzer** [comment]
I looks like your mismatching some principles here.

### Databases are how we store data, not how we view it.
Doctrine simple stores a value, in the datetime case it's persists the timestamp value stored inside datetime. Doctrine has no knowledge of locale or UTC, so Doctrine is never to blame.

Datetime's timestamp is UTC aware so it represents a timestamp independent of the locale your in. What you want.

### Viewing data
When viewing you have more options representing the time. The key to use the correct locale / timezone per domain.

example:
Meaning If I publish a dutch article at 18:00 in my current timezone, only the countries in my current timezone it we be displayed at 18:00. 

If I want to publish an English article at 18:00 in the USA, then in need to make sure that website domain lives in a different locale/timezone in order for it do be displayed correctly. The rest will work by itself.

**Comment by jeroendesloovere** [comment]
@Thijzer Thanks for your reply.
I actually do understand how everything is working, mayby my issue description above wasn't that clear...

Any solution suggestions?

**Comment by jeroendesloovere** [comment]
Didn't read the documentation entirely,
but it seems that [Carbon](http://carbon.nesbot.com/docs/) could help us.

**Comment by carakas** [comment]
http://docs.doctrine-project.org/en/latest/cookbook/working-with-datetime.html overwriting the datetime of doctrine with this might be a solution while on version 4.X.X

**Comment by carakas** [comment]
I'll fix this in a not BC Break way

**Comment by carakas** [comment]
the real issue here was that doctrine didn't store the datetimes in UTC but in the timezone dat was currently used in fork, its fixed in #1778

**Comment by jeroendesloovere** [comment]
Hoeray!
